[Magic WAN] strongSwan and VyOS IPsec config templates (#4038)
cloudflare/cloudflare-docs: content/magic-wan/tutorials/ipsec/strongswan.md

> ---

Reviewer: This should match what we do for the others.

pcx-content-type: tutorial
title: strongSwan
---

# strongSwan

> This tutorial contains a sample template of the `ipsec.conf`file for a working IPsec tunnel configuration established between a Linux machine running strongSwan and Cloudflare’s Magic service.

Reviewer: ```suggestion
This tutorial contains a sample template of the `ipsec.conf` file for a working IPsec tunnel configuration established between a Linux machine running strongSwan and Cloudflare’s Magic service.
```

## Configuration parameters

### Phase 1

- **Encryption**
    - AES-CBC with 256-bit key length
    - AES-GCM with 256-bit key length
- **Integrity**
    - SHA-256
- **Diffie-Hellman group**
    - DH group 14 (2048-bit MODP group)
- **PRF**
    - SHA-512

### Phase 2

- **Encryption**
    - AES with 256-bit key length
- **Integrity**
    - SHA-256
- **Diffie-Hellman group**
    - DH group 14 (2048-bit MODP group)

## Configuration template

```bash
config setup
    charondebug="all"
    uniqueids=yes

conn %default
    ikelifetime=4h
    rekey=yes
    reauth=no
    keyexchange=ikev2
    authby=secret
    dpdaction=restart
    closeaction=restart

conn <tunnel_name>
    auto=start
    mark = 50
    type=tunnel
    fragmentation=no
    leftauth=psk
    left=<IP_ADDR_OF_LINUX_UPLINK_TO_CF>
    leftid=<IPSEC_ID_STRING_IN_RESULT_OF_PSK_KEY-GEN_VIA_CF_API>
    leftsubnet=0.0.0.0/0
    right=<CF_ANYCAST_IP>
    rightid=<CF_ANYCAST_IP>
    rightsubnet=0.0.0.0/0
    rightauth=psk
    ike=aes256gcm16-prfsha512-modp2048
    esp=aes256gcm16-prfsha512-modp2048
    replay_window=0
```

### Dead Peer Detection (DPD)

> In the above `ipsec.conf `file in `conn %default` section, setting `dpdaction=restart` enables Dead Peer Detection (DPD) to actively check and re-establish IPsec tunnels in the event of communication timeouts. In addition, `closeaction=restart` is set to actively re-establish the tunnels in the event that the remote peer (i.e. Cloudflare Magic service) unexpectedly closes it.

Reviewer: ```suggestion
In the above `ipsec.conf` file in `conn %default` section, setting `dpdaction=restart` enables Dead Peer Detection (DPD) to actively check and re-establish IPsec tunnels in the event of communication timeouts. In addition, `closeaction=restart` is set to actively re-establish the tunnels in the event that the remote peer (usually a Cloudflare Magic service) unexpectedly closes it.
```
